client/components/backup-retention-management/retention-option-on-cancel-purchase/index.tsx

```diff
@@ -0,0 +1,164 @@
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+import { updateBackupRetention } from 'calypso/state/rewind/retention/actions';
+import { BACKUP_RETENTION_UPDATE_REQUEST } from 'calypso/state/rewind/retention/constants';
+import getActivityLogVisibleDays from 'calypso/state/rewind/selectors/get-activity-log-visible-days';
+import getBackupRetentionDays from 'calypso/state/rewind/selectors/get-backup-retention-days';
+import getBackupRetentionUpdateRequestStatus from 'calypso/state/rewind/selectors/get-backup-retention-update-status';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import LoadingPlaceholder from '../loading';
+import RetentionConfirmationDialog from '../retention-confirmation-dialog';
+import type { WithCamelCaseSlug, WithSnakeCaseSlug } from '@automattic/calypso-products';
+import type { RetentionPeriod } from 'calypso/state/rewind/retention/types';
+import './style.scss';
+
+interface BackupRetentionOptionOnCancelPurchaseProps {
+	purchase: WithCamelCaseSlug | WithSnakeCaseSlug;
+}
+
+const RetentionOptionOnCancelPurchase: React.FC = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const MINIMUM_RETENTION_TO_OFFER = 2;
+
+	const [ retentionOfferCardVisible, setRetentionOfferCardVisible ] = useState( false );
+	const [ confirmationDialogVisible, setConfirmationDialogVisible ] = useState( false );
+
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const siteSlug = useSelector( getSelectedSiteSlug );
+
+	// Retention period included in customer plan
+	const planRetentionPeriod = useSelector( ( state ) =>
+		getActivityLogVisibleDays( state, siteId )
+	);
+
+	// Retention period set by customer (if any)
+	const customerRetentionPeriod = useSelector( ( state ) =>
+		getBackupRetentionDays( state, siteId )
+	);
+
+	const updateRetentionRequestStatus = useSelector( ( state ) =>
+		getBackupRetentionUpdateRequestStatus( state, siteId )
+	);
+
+	// The retention days that currently applies for this customer.
+	const currentBackupRetention = customerRetentionPeriod || planRetentionPeriod || 0;
+	const updateRetentionPeriod = useCallback( () => {
+		dispatch(
+			recordTracksEvent( 'calypso_jetpack_backup_storage_retention_submit_click', {
+				retention_option: MINIMUM_RETENTION_TO_OFFER,
+			} )
+		);
+
+		dispatch( updateBackupRetention( siteId, MINIMUM_RETENTION_TO_OFFER as RetentionPeriod ) );
+	}, [ dispatch, siteId ] );
+
+	const handleUpdateRetention = useCallback( () => {
+		setConfirmationDialogVisible( true );
+	}, [] );
+
+	const onConfirmationClose = useCallback( () => {
+		setConfirmationDialogVisible( false );
+	}, [] );
+
+	useEffect( () => {
+		if (
+			updateRetentionRequestStatus === BACKUP_RETENTION_UPDATE_REQUEST.SUCCESS ||
+			updateRetentionRequestStatus === BACKUP_RETENTION_UPDATE_REQUEST.FAILED
+		) {
+			setConfirmationDialogVisible( false );
+			dispatch( {
```

Comment:
Similar to https://github.com/Automattic/wp-calypso/pull/75400/files#r1160850079